wa/workloads/exoplayer/__init__.py

```diff
@@ -0,0 +1,185 @@
+REGEXPS = {
+    'start'    : '.*Displayed com.google.android.exoplayer2.demo/.PlayerActivity',
+    'duration' : '.*period \[(?P<duration>[0-9]+.*)\]',
+    'end'      : '.*state \[.+, .+, E\]'
+}
+
+
+DOWNLOAD_URLS = {
+    'mp4_1080p': 'http://distribution.bbb3d.renderfarming.net/video/mp4/bbb_sunflower_1080p_30fps_normal.mp4',
+    'mov_720p':  'http://download.blender.org/peach/bigbuckbunny_movies/big_buck_bunny_720p_h264.mov',
+    'mov_480p':  'http://download.blender.org/peach/bigbuckbunny_movies/big_buck_bunny_480p_h264.mov',
+}
+
+
+class ExoPlayer(ApkWorkload):
+    """
+    Android ExoPlayer (YouTube)
+
+    ExoPlayer is the basic video player library that is used by the YouTube
+    android app. The aim of this workload is to test a proxy for YouTube
+    performance on targets where running the real YouTube app is not possible
+    due its dependencies.
+
+    ExoPlayer sources: https://github.com/google/ExoPlayer
+
+    The 'demo' application is used by this workload.  It can easily be built by
+    loading the ExoPlayer sources into Android Studio.
+
+    Version r2.4.0 built from commit d979469 is known to work
+    """
+
+    name = 'exoplayer'
+
+    video_directory = os.path.join(settings.dependencies_directory, name)
+
+    versions = ["2.4.0"]
+    action = 'com.google.android.exoplayer.demo.action.VIEW'
+
+    parameters = [
+        Parameter('package_name', default='com.google.android.exoplayer2.demo', override=True),
+        Parameter('version', allowed_values=versions, default=versions[-1], override=True),
+        # The app requests runtime permissions, which AFAICT you can't grant via
+        # "pm grant", only "adb install -g". Therefore we'll just have to
+        # reinstall every time.
+        Parameter('force_install', default=True, override=True),
+
+        Parameter('play_duration', kind=int, default=20,
+                  description='Playback duration of the video file. This become the duration of the workload.'),
+        Parameter('format', default='mp4_1080p', allowed_values=DOWNLOAD_URLS.keys(),
+                  description='Specifies which format video file to play.'),
+        Parameter('filename',
+                  description="""
+                   The name of the video file to play. This can be either a path
+                   to the file anywhere on your file system, or it could be just a
+                   name, in which case, the workload will look for it in
+                   ``{}``
+                   *Note*: either format or filename should be specified, but not both!
+                  """.format(video_directory)),
+        Parameter('force_dependency_push', kind=boolean, default=False,
+                  description="""
+                  If true, video will always be pushed to device, regardless
+                  of whether the file is already on the device.  Default is ``False``.
+                  """),
+    ]
+
+    @property
+    def host_video_file(self):
```

Comment:
This should not be a property.  `selected_file ` shoud probably be resolved as part of `init_resource`.

Reply:
Hmm true. I just copied this bit from the 'video' workload, will change it.